### ISSUE TITLE
A question about model_decoder.py

### ISSUE THREAD

**Issue by Alivewwt** [dantegates/attention-is-all-you-need#2]
your code is very great! but i have a question about your model, how can i use pre-trained word vectors for you model? when i used pre-trained word vectors ,the dimension of word embeedings is set to 300, the model would run errors. How to slove this problem.
Thanks in advance.

**Comment by cadent-dgates** [comment]
Currently this isn't possible, however it would be fairly easy to modify the code to do so.

Assuming you plan on passing the word vectors as input to `model.fit()` (as opposed to using a pre-trained model - which could also be done in a similar way) you could add a parameter to `TransformerDecoder.__init__()` indicating that the model should not instantiate an `Embedding` layer. You would also need to add as parameters the dimensions `d_k` and `d_v` and choose them appropriately to work with the dimension of your word vectors. Lastly you would have to omit the code that uses the word embeddings as the weights of the final linear transformation before the softmax (which, by the way, seems to be the default setting in the tensor2tensor implementation)

I don't currently plan on adding anything like this as I read that experiments showed no benefit in using pre-trained vectors over learning the embeddings as part of the Transformer model.

You could fork my repo and add this feature yourself, and if the code is clean and you'd like to do a pull request I would be open to collaborations.

**Comment by dantegates** [comment]
Currently this isn't possible, however it would be fairly easy to modify the code to do so.

Assuming you plan on passing the word vectors as input to model.fit() (as opposed to using a pre-trained model - which could also be done in a similar way) you could add a parameter to TransformerDecoder.__init__() indicating that the model should not instantiate an Embedding layer. You would also need to add as parameters the dimensions d_k and d_v and choose them appropriately to work with the dimension of your word vectors. Lastly you would have to omit the code that uses the word embeddings as the weights of the final linear transformation before the softmax (which, by the way, seems to be the default setting in the tensor2tensor implementation)

I don't currently plan on adding anything like this as I read that experiments showed no benefit in using pre-trained vectors over learning the embeddings as part of the Transformer model.

You could fork my repo and add this feature yourself, and if the code is clean and you'd like to do a pull request I would be open to collaborations.
